frontend-react/src/network/hooks/useNetwork.ts

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+
+import { Endpoint, Api } from "../api/Api";
+
+export interface ResponseType<T> {
+    loading: boolean;
+    data?: T;
+    status: number;
+    message: string;
+}
+
+export function useNetwork<T>({ url, api }: Endpoint): ResponseType<T> {
+    const [response, setResponse] = useState<ResponseType<T>>({
+        loading: true,
+        data: undefined,
+        status: 0,
+        message: "",
+    });
+
+    useEffect(() => {
+        /* Fetch data and handle any parsing needed */
+        axios
```

Comment:
Can this call ever throw? Maybe test going offline in the browser and seeing what happens?

Reply:
The effect hook wouldn't throw but the `axios` call would, and the catch block handles that for us.